pandas/tseries/tests/test_timeseries.py

```diff
@@ -743,6 +743,22 @@ def test_to_datetime_unit(self):
         result = to_datetime(s,unit='s')
         expected = Series([ Timestamp('2013-06-09 02:42:28') + timedelta(seconds=t) for t in range(20) ] + [NaT])
         assert_series_equal(result,expected)
+        
+    def test_to_datetime_origin(self):
+        # for origin as julian
+        julian_dates = pd.date_range('2014-1-1', periods=10).to_julian_date().values
+        result = Series(pd.to_datetime(julian_dates, unit='D', origin='julian'))
+        expected = Series(pd.to_datetime(julian_dates - pd.Timestamp(0).to_julian_date(), unit='D'))
+        assert_series_equal(result,expected)
+        
+        # for origin as 1960-01-01
+        EPOCH1960 = date(1960, 1, 1)
+        daysFromEpoch = range(5)
+        result = Series(pd.to_datetime(daysFromEpoch, unit='D', origin=EPOCH1960))
+        expected = Series(pd.date_range('1960-1-1', periods=5))
+        assert_series_equal(result,expected)
+        
```

Comment:
test with origin as something else as well (e.g. a different string/timestamp convertible), use a loop to pass ``date,datetime,Timestamp,np.datetime64,string`` types (these are all ``Timestamp`` convertible).